packages/components/src/ui/heading/hook.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { useContextSystem } from '@wp-g2/context';
+import { getHeadingFontSize, ui } from '@wp-g2/styles';
+import type { ViewOwnProps } from '@wp-g2/create-styles';
+// eslint-disable-next-line no-restricted-imports
+import type { CSSProperties } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { Props as TextProps, TextSize } from '../text/types';
+import { useText } from '../text';
+
+export type HeadingSize = 1 | 2 | 3 | 4 | 5 | 6;
+
+export interface HeadingProps extends TextProps {
+	/**
+	 * `Heading` will typically render the sizes `1`, `2`, `3`, `4`, `5`, or `6`, which map to `h1`-`h6`. However, it can render any size, including non `px` values.
+	 *
+	 * @default 3
```

Comment:
In the block editor, the Heading block defaults to size 2. It's also called `level` in attributes rather than `size`. I'm curious why would it be different here.

Reply:
Calling it `level` makes sense to me and would remove the issues with shadowing the `size` prop from Text (which we should probably just exclude from `Heading`. If you want some text with a specific size, just use `Text` not `Heading` which should be reserved specifically for the `h1-6` elements.

It would also allow us to use the semantic elements (unless `as` is passed in) rather than `div`.

/cc @ItsJonQ

Reply:
I'm down with changing the prop to `level` 🙌 

> If you want some text with a specific size, just use Text not Heading which should be reserved specifically for the h1-6 elements.

V good point!